### PR TITLE
Adam implementation minor fix

### PR DESCRIPTION
**Pull request by farhadrgh**
This PR is in accordance with #22628 
I had submitted the PR for `adam.py` and `adamw.py` but had forgotten about the `adam.cpp`.

### CODE REVIEW

**Review by yf225 — APPROVED** [review]

**Comment by yf225** [comment]
Thanks for the fix @farhadrgh!

File: torch/csrc/api/src/optim/adam.cpp [hunk 1]
```diff
@@ -31,24 +31,24 @@ void Adam::step() {
...
+        1 - std::pow(options.beta1_, buffer_at(step_buffers, i));
+    const auto bias_correction2 =
+        1 - std::pow(options.beta2_, buffer_at(step_buffers, i));
 
     exp_average.mul_(options.beta1_).add_(p.grad(), 1 - options.beta1_);
     exp_average_sq.mul_(options.beta2_)
         .addcmul_(p.grad(), p.grad(), 1 - options.beta2_);
 
-    Tensor denom = exp_average_sq;
+    Tensor denom = exp_average_sq / bias_correction2;
```
Comment (SsnL) [inline torch/csrc/api/src/optim/adam.cpp]: arguably this should be in an `else` block to avoid computation when `options.amsgrad_`

Comment (vincentqb) [inline torch/csrc/api/src/optim/adam.cpp]: I believe `bias_correction2`  should have a square root, isn't that so?

**Review by vincentqb — CHANGES_REQUESTED** [review]
Please confirm that line 43 is indeed correct.

File: torch/csrc/api/src/optim/adam.cpp (hunk 1 above)
Comment (farhadrgh) [reply torch/csrc/api/src/optim/adam.cpp]: Sorry, I forgot to remove the `sqrt` in this line. I fixed it in 72c6582.
We are doing the `denom.sqrt()` at the end of `adam.cpp` whereas in python scripts the square root is calculated before `addcdiv_ `

Comment (farhadrgh) [reply torch/csrc/api/src/optim/adam.cpp]: I changed declaration in 72c6582.

**Review by vincentqb — APPROVED** [review]
Thanks, this looks good to me!

**Pull request closed without merging**